File: app/src/main/java/org/wycliffeassociates/converter/AnalyserTask.java
package org.wycliffeassociates.converter;

import android.os.AsyncTask;
import android.support.v4.app.Fragment;

/**
 * Created by mXaln on 11/11/17.
 */

public final class AnalyserTask extends AsyncTask<Void, Integer, Void> {

    AnaliserResultCallback callback;

    public AnalyserTask(Fragment c) {
        this.callback = (AnaliserResultCallback) c;
    }

    @Override
    protected Void doInBackground(Void... params) {
        callback.startAnalyze();
        return null;
    }

    @Override
    protected void onPreExecute() {
        callback.analyzeStarted();
        super.onPreExecute();
    }

    @Override
    protected void onProgressUpdate(Integer... values) {
        super.onProgressUpdate(values);
    }

    @Override
    protected void onPostExecute(Void param) {
        callback.analyzeDone();
        super.onPostExecute(param);
    }

    public interface AnaliserResultCallback {
        Void startAnalyze();
        Void analyzeStarted();
        Void analyzeDone();
    }
}

File: app/src/main/java/bible/translationtools/converter/AnalyserTask.kt
package bible.translationtools.converter

import android.os.AsyncTask
import androidx.fragment.app.Fragment

/**
 * Created by mXaln on 11/11/17.
 */
class AnalyserTask(c: Fragment?) : AsyncTask<Void?, Int?, Void?>() {
    var callback: AnalyzerResultCallback

    init {
        this.callback = c as AnalyzerResultCallback
    }

    override fun doInBackground(vararg params: Void?): Void? {
        callback.startAnalyze()
        return null
    }

    override fun onPreExecute() {
        callback.analyzeStarted()
        super.onPreExecute()
    }

    override fun onProgressUpdate(vararg values: Int?) {
        super.onProgressUpdate(*values)
    }

    override fun onPostExecute(param: Void?) {
        callback.analyzeDone()
        super.onPostExecute(param)
    }

    interface AnalyzerResultCallback {
        fun startAnalyze()
        fun analyzeStarted()
        fun analyzeDone()
    }
}
